03. An EDSL for Hard-working IT Programmers.md

```diff
@@ -1,145 +1,172 @@
+## "Deconstructing the specification" recipe considered harmful ##
 
-In any case, in OOP, the programs are made of disconnected pieces, the software does not follow the natural flow as it would have been naturally extracted from the specifications and there are low level details. There will appear functionally uninteresting elements at the top level of the design, so the resulting code is hard to maintain.
+Usually the specification of some task can naturally be expressed as a process of step-by-step imperatives (often using verbs in second-person, active tense). People connect the relevant elements directly, without concern for other secondary elements. For example, a recipe would say, "You fry the eggs." It would not say, "There is fire and there are eggs; you start the fire and the eggs are fried by the fire." You see that a complete description in terms of active and passive elements implies low level elements that you are not interesting when writing a specification. But in OOP development, these elements must appear in the form of objects. Creating an OOP solution requires the deconstruction of the specification recipe into multiple "third-person-passive" perspectives (e.g. "it was fried" instead of "fry it"), one for each class or object that alternatively act as passive and active elements.
 
-The application/service created with this deconstruction is not composable. There is no way to insert your service within something bigger only with a single invocation. You can not even do it with the objects that you fully control, since an object, by definition, is a box with many connectors, so it can not be assembled in a pipeline. This derives in the scarce reusability of the software, and the need of profuse documentation. It is inelegant, buggy, hard to maintain, and permits an huge number of arbitrary alternatives in the design space that aggravates the mentioned problems. It may be though that this is good for the IT business, but not in the medium-long term. There are better ways to do it.
+In OOP, you can not create a function or method called `fry-the-eggs` that stops the fire when the eggs are fried, unless you want to block the execution of everything else. So you need to manage explicitly at least two threads that sooner or later will have to communicate asynchronously. Alternatively, you may define `start-fire`, `stop-fire`, and a callback called `egg-fried` so that you wire up the application with these elements. There is no way to express `fry-the-eggs` in a single, self-contained expression that you can reuse. As a consequence, OOP can not produce composable programs.
```

Comment:
I think I'm starting to get it, but I'm not sure why `fro-the-eggs` must be a blocking call?